kemeny_sourcemash.tex

```diff
@@ -59,15 +59,41 @@ \subsection{Development: localhost}
 
 \section{Technical Description}
 
-While 
+Sourcemash was built to be used as an RSS feed reader, but at its core is a categorization algorithm. We will break down the technical description into these two components: the categorization algorithm and the RSS reader that incorporates the algorithm to allow users to get parse through the news faster.
 
-\subsection{Subsection 1}
+\subsection{Categorization Algorithm}
 
-When
+Because we need articles to group together by category, it was import that our categorization algorithm generate categories with a decent number of overlapping categories. In addition, it was important that the algorithm work quickly because it would need to process many articles 24/7 to stay up-to-date with the latest news.
 
-\subsection{Subsection 2}
+To help with the first concern, we needed a way to normalize our category data. This meant that capitalization, punctuation, and pluralization should all be consistent for each instance of a given category. Upon researching state-of-the-art categorization techniques, we came across the CommunityCluster algorithm\cite{Grineva}, which uses Wikipedia lookups to generate possible keywords from an article. By using Wikipedia articles, categories would be normalized by converting n-grams from a source article into official Wikipedia article titles.
 
-When
+For the second concern, we sped up the CommunityCluster algorithm by only running it on the n-most-common ngrams in each feed article. This would require less lookups using the Wikipedia API.
```

Comment:
For the efficiency concern, we increased the speed of the CommunityCluster algorithm by pre-processing article data and running the algorithm only on the n-most-common n-grams in each feed article. This requires less lookups using the Wikipedia API, which is the time limiting constraint in the algorithm's production implementation.